server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1707,6 +1707,10 @@ public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineExcepti
                         logger.trace("starting commit for flush; commitTranslog=true");
                         commitIndexWriter(indexWriter, translog, null);
                         logger.trace("finished commit for flush");
+
+                        // a temporary debugging to investigate test failure - issue#32827. Remove when the issue is resolved
+                        logger.debug("new commit on flush");
```

Comment:
It might help to know which of the conditions (`indexWriter.hasUncommittedChanges()`, `force` or `shouldPeriodicallyFlush()`) triggered this flush  to help us work out what is causing this?